src/taco-remote-lib/ios/ios.ts

```diff
@@ -80,23 +83,30 @@ class IOSAgent implements ITargetPlatform {
         var appLaunchStepTimeout = this.appLaunchStepTimeout;
         var cfg: utils.CordovaConfig = utils.CordovaConfig.getCordovaConfig(buildInfo.appDir);
         iosAppRunner.startDebugProxy(proxyPort)
-            .then(function (nativeProxyProcess: child_process.ChildProcess): Q.Promise<net.Socket> {
-            return iosAppRunner.startApp(cfg.id(), proxyPort, appLaunchStepTimeout);
-        }).then(function (success: net.Socket): void {
-            res.status(200).send(buildInfo.localize(req, resources));
-        }, function (failure: any): void {
-            if (failure.message) {
-                var response = resources.getStringForLanguage(req, failure.message);
-                if (!response) {
-                    response = resources.getStringForLanguage(req, "UnableToLaunchAppWithReason", failure.message);
+            .then(function(nativeProxyProcess: child_process.ChildProcess): Q.Promise<net.Socket> {
+                return iosAppRunner.startApp(cfg.id(), proxyPort, appLaunchStepTimeout, (isCrash) => {
+                    // The app has terminated
+                    if (isCrash) {
+                        buildInfo.updateStatus(BuildInfo.CRASHED);
+                    } else {
+                        buildInfo.updateStatus(BuildInfo.TERMINATED);
+                    }
+                });
+            }).then(function(success: net.Socket): void {
+                res.status(200).send(buildInfo.localize(req, resources));
+            }, function(failure: any): void {
+                if (failure.message) {
```

Comment:
All this failure message processing can be extracted in a different method. Would we be able to reuse it for any other cases?